### ISSUE TITLE
Resolve RUSTSEC-2020-0146

### ISSUE THREAD

**Issue by michaelsproul** [sigp/lighthouse#2237]
## Description

Our current version of `libp2p` depends transitively on `generic-array 0.12.3` which is vulnerable https://rustsec.org/advisories/RUSTSEC-2020-0146

**Comment by michaelsproul** [comment]
Updating libp2p alone doesn't help, `libsecp256k1` and `warp` are also problematic (even with #2238 merged)

```
generic-array v0.12.3
    ├── block-buffer v0.7.3
    │   ├── sha-1 v0.8.2
    │   │   └── headers v0.3.3
    │   │       ├── warp v0.3.0
    │   │       │   ├── http_api v0.1.0
    │   │       │   │   └── client v0.2.0
    │   │       │   │       └── beacon_node v1.1.3
    │   │       │   │           ├── boot_node v1.1.3
    │   │       │   │           │   └── lighthouse v1.1.3
    │   │       │   │           ├── lighthouse v1.1.3 (*)
    │   │       │   │           └── node_test_rig v0.2.0
    │   │       │   │               ├── (dev) beacon_node v1.1.3 (*)
    │   │       │   │               └── simulator v0.2.0
    │   │       │   ├── http_metrics v0.1.0
    │   │       │   │   └── client v0.2.0 (*)
    │   │       │   ├── validator_client v0.3.5
    │   │       │   │   ├── lighthouse v1.1.3 (*)
    │   │       │   │   ├── node_test_rig v0.2.0 (*)
    │   │       │   │   └── simulator v0.2.0 (*)
    │   │       │   └── warp_utils v0.1.0
    │   │       │       ├── http_api v0.1.0 (*)
    │   │       │       ├── http_metrics v0.1.0 (*)
    │   │       │       └── validator_client v0.3.5 (*)
    │   │       └── warp_utils v0.1.0 (*)
    │   └── sha2 v0.8.2
    │       └── libsecp256k1 v0.3.5
    │           ├── eth2 v0.1.0
    │           │   ├── account_manager v0.3.5
    │           │   │   └── lighthouse v1.1.3 (*)
    │           │   ├── beacon_chain v0.2.0
    │           │   │   ├── beacon_node v1.1.3 (*)
    │           │   │   ├── client v0.2.0 (*)
    │           │   │   ├── (dev) fork_choice v0.1.0
    │           │   │   │   ├── beacon_chain v0.2.0 (*)
    │           │   │   │   └── http_api v0.1.0 (*)
    │           │   │   ├── http_api v0.1.0 (*)
    │           │   │   ├── http_metrics v0.1.0 (*)
    │           │   │   ├── network v0.2.0
    │           │   │   │   ├── client v0.2.0 (*)
    │           │   │   │   ├── http_api v0.1.0 (*)
    │           │   │   │   └── slasher_service v0.1.0
    │           │   │   │       └── client v0.2.0 (*)
    │           │   │   ├── slasher_service v0.1.0 (*)
    │           │   │   ├── timer v0.2.0
    │           │   │   │   └── client v0.2.0 (*)
    │           │   │   └── warp_utils v0.1.0 (*)
    │           │   ├── eth1 v0.2.0
    │           │   │   ├── beacon_chain v0.2.0 (*)
    │           │   │   ├── client v0.2.0 (*)
    │           │   │   ├── genesis v0.2.0
    │           │   │   │   ├── beacon_chain v0.2.0 (*)
    │           │   │   │   ├── beacon_node v1.1.3 (*)
    │           │   │   │   ├── client v0.2.0 (*)
    │           │   │   │   ├── lcli v1.1.3
    │           │   │   │   ├── (dev) network v0.2.0 (*)
    │           │   │   │   └── node_test_rig v0.2.0 (*)
    │           │   │   ├── http_api v0.1.0 (*)
    │           │   │   └── simulator v0.2.0 (*)
    │           │   ├── http_api v0.1.0 (*)
    │           │   ├── http_metrics v0.1.0 (*)
    │           │   ├── node_test_rig v0.2.0 (*)
    │           │   ├── validator_client v0.3.5 (*)
    │           │   └── warp_utils v0.1.0 (*)
    │           ├── libp2p-core v0.27.1
    │           │   ├── discv5 v0.1.0-beta.3
    │           │   │   ├── eth2_libp2p v0.2.0
    │           │   │   │   ├── beacon_node v1.1.3 (*)
    │           │   │   │   ├── boot_node v1.1.3 (*)
    │           │   │   │   ├── client v0.2.0 (*)
    │           │   │   │   ├── eth2 v0.1.0 (*)
    │           │   │   │   ├── http_api v0.1.0 (*)
    │           │   │   │   ├── http_metrics v0.1.0 (*)
    │           │   │   │   ├── lcli v1.1.3 (*)
    │           │   │   │   ├── network v0.2.0 (*)
    │           │   │   │   └── slasher_service v0.1.0 (*)
    │           │   │   └── (dev) http_api v0.1.0 (*)
    │           │   ├── libp2p v0.35.1
    │           │   │   └── eth2_libp2p v0.2.0 (*)
    │           │   ├── libp2p-dns v0.27.0
    │           │   │   └── libp2p v0.35.1 (*)
    │           │   ├── libp2p-gossipsub v0.28.0
    │           │   │   └── libp2p v0.35.1 (*)
    │           │   ├── libp2p-identify v0.27.0
    │           │   │   └── libp2p v0.35.1 (*)
    │           │   ├── libp2p-mplex v0.27.1
    │           │   │   └── libp2p v0.35.1 (*)
    │           │   ├── libp2p-noise v0.29.0
    │           │   │   └── libp2p v0.35.1 (*)
    │           │   ├── libp2p-swarm v0.27.2
    │           │   │   ├── libp2p v0.35.1 (*)
    │           │   │   ├── libp2p-gossipsub v0.28.0 (*)
    │           │   │   └── libp2p-identify v0.27.0 (*)
    │           │   ├── libp2p-tcp v0.27.1
    │           │   │   └── libp2p v0.35.1 (*)
    │           │   ├── libp2p-websocket v0.28.0
    │           │   │   └── libp2p v0.35.1 (*)
    │           │   └── libp2p-yamux v0.30.1
    │           │       └── libp2p v0.35.1 (*)
    │           └── validator_client v0.3.5 (*)
    ├── crypto-mac v0.7.0
    │   └── hmac v0.7.1
    │       └── hmac-drbg v0.2.0
    │           └── libsecp256k1 v0.3.5 (*)
    ├── digest v0.8.1
    │   ├── hmac v0.7.1 (*)
    │   ├── hmac-drbg v0.2.0 (*)
    │   ├── libsecp256k1 v0.3.5 (*)
    │   ├── sha-1 v0.8.2 (*)
    │   └── sha2 v0.8.2 (*)
    └── hmac-drbg v0.2.0 (*)
```

Dep graph generated using `cargo deny check bans` and this `deny.toml`:

```toml
[bans]
multiple-versions = "allow"

[[bans.deny]]
name = "generic-array"
version = "=0.12.3"
```

**Comment by michaelsproul** [comment]
`generic-array` has released a patched `0.12.4` version which we can upgrade to, PR incoming
